nsqd/http.go

```diff
@@ -739,3 +754,22 @@ func getOptByCfgName(opts interface{}, name string) (interface{}, bool) {
 	}
 	return nil, false
 }
+
+func (s *httpServer) startDraining(w http.ResponseWriter, req *http.Request, ps httprouter.Params) (interface{}, error) {
+	go func() {
+		// in some cases StartDraining results in an exit immediately
+		// allow this API call to respond before exiting
+		time.Sleep(time.Millisecond)
```

Comment:
Since this is a HTTP request, there is some desire to allow the HTTP request to complete (i.e. the caller should get a HTTP response back) before the action takes place. If we call *immediately* an shutdown can (and will in many cases) close this connection and the caller will get a connection error.

I'm open to other ways to solve this (because it's obviously not a generally good pattern to rely on timing), but this seemed reasonable for the scope of this endpoint.